components/raftstore/src/store/entry_storage.rs

```diff
@@ -980,8 +1060,127 @@ impl<ER: RaftEngine> EntryStorage<ER> {
         self.last_term = last_term;
     }
 
+    pub fn entry_cache_warmup_state(&self) -> &Option<CacheWarmupState> {
+        &self.cache_warmup_state
+    }
+
+    pub fn entry_cache_warmup_state_mut(&mut self) -> &mut Option<CacheWarmupState> {
+        &mut self.cache_warmup_state
+    }
+
+    pub fn exit_entry_cache_warmup_state(&mut self) {
+        self.cache_warmup_state = None;
+    }
+
+    /// Trigger a task to warm up the entry cache.
+    ///
+    /// This will ensure the range [low..high..last_index] are loaded into
+    /// cache. Return the high index of the warmup range if a task is
+    /// successfully triggered.
+    pub fn async_warm_up_entry_cache(&mut self, low: u64) -> Option<u64> {
+        let high = if let Some(first_index) = self.entry_cache_first_index() {
+            if low >= first_index {
+                // Already warmed up.
+                self.cache_warmup_state = Some(CacheWarmupState::new());
```

Comment:
why cache_warmup_state needs to be set as it will be cleared anyway after the function returns.

Reply:
The state is not cleared after the function returns. It is only cleared in two cases
1. the peer becomes leader
2. the state becomes stale